Rakefile

```diff
@@ -50,20 +52,36 @@ namespace :arm do
     end
   end
 
-  desc 'Regen code for each of the Azure Resource Manager projects'
-  task :regen => :clean_generated do
+  desc 'Regen code for each sdk with all its api versions'
+  task :apiVersions => :clean_generated_api_versions do
+    json = get_config_file
     REGEN_EXCLUDES = ['azure_sdk']
-    each_gem do |dir|
+    each_gem do |dir| # dir corresponds to each azure_mgmt_* folder
       unless REGEN_EXCLUDES.include?(dir.to_s)
         puts "\nGenerating #{dir}\n"
-        md = REGEN_METADATA[dir.to_sym]
         ar_base_command = "#{OS.windows? ? '' : 'mono '} #{REGEN_METADATA[:autorest_loc]}"
-
-        command = "#{ar_base_command} #{md[:spec_uri]} --package-version=#{md[:version]} --namespace=#{md[:ns]} --package-name=#{md[:pn].nil? ? dir : md[:pn]} --output-folder=#{File.join(Dir.pwd, 'lib')} --ruby --azure-arm"
-        command += " --#{md[:options]}" unless md[:options].nil?
-        command += " --tag=#{md[:tag]}" unless md[:tag].nil?
-
-        execute_and_stream(command)
+        md = json[dir] # there should be an entry in the metadata for each of the api versions to generate
+        puts "md: #{md}"
+        package_name = dir
+        md.each do |api_version_pkg, api_version_value|
+         ar_arguments = ''
+         output_folder = ''
+         api_version_value.each do |argument_name, argument_value|
+           if argument_name.casecmp("output-folder") == 0
+             output_folder = argument_value
+           else
+             if argument_name.casecmp("markdown") == 0
+               ar_arguments = ar_arguments + " #{argument_value}"
+             else
+               ar_arguments = ar_arguments + " --#{argument_name}=#{argument_value}"
+             end
+           end
+         end
+         # puts "Arguments: #{ar_arguments}"
+         command = "#{ar_base_command} --package-name=#{package_name} #{ar_arguments} --output-folder=#{File.join(Dir.pwd, 'lib', output_folder)} --ruby --azure-arm"
+         puts "command: #{command}"
```

Comment:
minor: we can remove this line as `execute_and_stream` would print the command it is running..

Reply:
yes, I can remove most of the puts :) I added them for my debugging purposes and just left them in for now.